navoica_api/api/v1/views.py

```diff
@@ -174,3 +188,113 @@ def calculate_progress():
                          "completion_value": calculated_progress}
 
         return Response(response_dict, status=status.HTTP_200_OK)
+
+
+class CertificatesListView(ListAPIView):
+    """
+        **Use Case**
+
+            * Get the list of a generated certificate for a specific course_id
+
+        **Example Request**
+
+            GET /api/certificates/v0/certificates/courses/{course_id}
+
+        **GET Parameters**
+
+            A GET request must include the following parameters.
+
+            * course_id: A string representation of a Course ID.
+
+        ** Query parameters**
+
+            * attachment: return ['Content-Disposition'] = 'attachment; filename={} response
+
+            * filters/search/ordering attributes: ('user__profile__name', 'user__username',
+            'user__email', 'created_date',)
+
+        **Response Values**
+
+            If the request for information about the user is successful, an HTTP 200 "OK" response
+            is returned.
+
+            The HTTP 200 response has the following values:
+
+            * count: The number of mappings for the backend.
+
+            * next: The URI to the next page of the mappings.
+
+            * previous: The URI to the previous page of the mappings.
+
+            * num_pages: The number of pages listing the mappings.
+
+            * results:  A list of mappings returned. Each collection in the list
+            contains these fields.
+
+                * profile_name: student lastname and surname
+
+                * username: student username
+
+                * email: student e-mail
+
+                * created_date: date of created certificate
+
+                * grade: grade of certificate
+    """
+
+    authentication_classes = (authentication.OAuth2AuthenticationAllowInactiveUser,
+                              authentication.SessionAuthenticationAllowInactiveUser,
+                              JwtAuthentication,)
+    permission_classes = (IsAuthenticated, IsCourseStaffInstructorOrStaff)
+    pagination_class = DefaultPagination
+    serializer_class = GeneratedCertificateSerializer
+    renderer_classes = (JSONRenderer, BrowsableAPIRenderer)
+    filter_backends = (OrderingFilter, DjangoFilterBackend, SearchFilter)
+    filterset_fields = search_fields = ordering_fields = ('user__profile__name', 'user__username',
+        'user__email', 'created_date',)
+
+    def __init__(self):
+        super(CertificatesListView, self).__init__()
+        self.course_id = None
```

Comment:
Ja tam lubię ustawiać to odrazu jako atrybut w klasie zamiast w init (mniej pisania :)) chyba że planujesz to dalej rozwijać

Reply:
pylint wyrzucał mi tutaj warning, więc mogłem albo tak, albo jako atrybut klasowy poza metodami klasowymi. Uznałem że tak powinno być ładniej.